### PR TITLE
[ci-skip] Skip ambassador installation when installed

### PR DESCRIPTION
**Pull request by adityajoshi12**
**Changelog**
This PR has changeset for skipping ambassador installation when it's installed. Optionally, now namespace can also be provided when installing the ambassador.

 

**Reviewed by**
@developer_github_id

 

**Linked issue**
#2111

### CODE REVIEW

File: platforms/shared/configuration/roles/setup/ambassador/tasks/main.yaml [hunk 1]
```diff
@@ -7,6 +7,23 @@
...
+- name: Set proxy namespace
+  set_fact:
+    proxy_namespace: "{{ network.env.namespace | default('default') }}"
+
+- name: Check if ambassador installed
+  k8s_info:
+    api_version: v1
+    kind: Service
+    name: ambassador
+    kubeconfig: { { kubeconfig_path } }
```
Comment (jagpreetsinghsasan) [inline platforms/shared/configuration/roles/setup/ambassador/tasks/main.yaml:19]: Please rename this to ``{{ kubeconfig_path }}`` and similarly to the next line var

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]
We shouldn't skip ambassador installation this simply.
There can be scenarios when there are 2 deployments using the same ambassador, but one ambassador is using a port range from 15010-15020 and then another deployment want to use next 10 ports (so they can simply mention 15010-15030 and ambassador will open those 10 more ports).
Either we check the opened ports are same as well to skip ambassador installation or we just include the namespaced proxy installation feature of this PR.

**Comment by jagpreetsinghsasan** [comment]
Also, try to use the [newer format](https://github.com/hyperledger/bevel/blob/develop/.github/pull_request_template.md) for PR message (and similarly the commit message) (most likely you will start receiving the new PR format once the develop branch is merged into main).

**Comment by adityajoshi12** [comment]
> We shouldn't skip ambassador installation this simply. There can be scenarios when there are 2 deployments using the same ambassador, but one ambassador is using a port range from 15010-15020 and then another deployment want to use next 10 ports (so they can simply mention 15010-15030 and ambassador will open those 10 more ports). Either we check the opened ports are same as well to skip ambassador installation or we just include the namespaced proxy installation feature of this PR.

Thanks for the suggestion, made the changes as per the suggestion. We are now checking if the ports are already opened in the existing installation. if they are found to be present then skip installation else the ambassador.

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]
There are still some inconsistencies within the code.
It does certain operations only when `no ambassador_installed`, for example: the task `Format ambassador ports`. This should still format ports, even if ambassador is installed.

Also once fixed, please create an issue to address this update in other platforms also (one issue to mention that we need to update this for fabric, corda, quorum, indy), so that we can have a track of it.

**Comment by jagpreetsinghsasan** [comment]
@adityajoshi12 I saw you added commit(s) after my last review. If you think its ready for review again, you should re-request for review 
image

File: platforms/shared/configuration/roles/setup/ambassador/tasks/main.yaml [hunk 2]
```diff
@@ -99,59 +123,122 @@
...
+    do
+      if [[ ! ${availablePortRange[*]} =~ ${i} ]]; then
+        echo "false" && exit 0
+      fi
+    done
+    echo "true"
+  register: ports_available
+  when:
+    - network.type != 'indy'
+    - ambassador_installed
```
Comment (jagpreetsinghsasan) [inline platforms/shared/configuration/roles/setup/ambassador/tasks/main.yaml:155]: Please have a space between tasks

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM, thankyou!

File: platforms/shared/configuration/roles/setup/ambassador/tasks/main.yaml (hunk 2 above)
Comment (suvajit-sarkar) [reply platforms/shared/configuration/roles/setup/ambassador/tasks/main.yaml:155]: @adityajoshi12 kindly have a look

Comment (adityajoshi12) [reply platforms/shared/configuration/roles/setup/ambassador/tasks/main.yaml:155]: @suvajit-sarkar updated...

**Pull request merged**